pkg/randomnumbergenerator/uuid.go

```diff
@@ -0,0 +1,13 @@
+package randomnumbergenerator
+
+import (
+	"github.com/google/uuid"
+)
+
+// UUIDGenerator implements the RandomNumberGenerator interface
+type UUIDGenerator struct{}
+
+// NewRandom returns a sufficiently random UUID
+func (u *UUIDGenerator) NewRandom() (uuid.UUID, error) {
+	return uuid.NewRandom()
```

Comment:
suggestion: If you used UUID1 then you could squeeze in a little more metadata in the strings. Ideally the contract is that the IDs are opaque strings. However, you could offer a selection of strategies to the user so they could control the implementation. At the very least, UUID1s will naturally sort in time order which can be a convenience if you're looking at a bunch of them.

Reply:
I don't see a future need to have these sort in any particular order, as even the future async status check API will be based on having a specific ID to check.

Further, I will absolutely not rely on the time order sorting if that is and implementation detail and not part of the defined interface.

Besides all of this, the reason this is an interface at all, and not just a function call is so that I could test the error path in my handler for the infinitesimal likelihood that a stateful call to generate a random number actually fails. I see no reason whatsoever to add the complexity of configurable UUID generation strategies to this service.